'pointer' is deprecated / 'const_pointer' is deprecated  clang 21.1.0

Good evening.
I'm using parallel-hashmap in R package tokenizers.bpe (https://cran.r-project.org/web/packages/tokenizers.bpe/index.html), source code at https://github.com/bnosac/tokenizers.bpe. 

The CRAN maintainers have currently upgraded their test suite on fedora to use clang version 21.1.0 (full compiler details [here](https://www.[stats.ox.ac.uk/pub/bdr/Rconfig/r-devel-linux-x86_64-fedora-clang](https://www.stats.ox.ac.uk/pub/bdr/Rconfig/r-devel-linux-x86_64-fedora-clang))). 
This gives a warning message on the use of pointer and const_pointer at 
https://github.com/bnosac/tokenizers.bpe/blob/master/src/parallel_hashmap/phmap_base.h#L1204
https://github.com/bnosac/tokenizers.bpe/blob/master/src/parallel_hashmap/phmap_base.h#L1207

> Found the following significant warnings:
>   ./parallel_hashmap/phmap_base.h:1204:1: warning: 'pointer' is deprecated [-Wdeprecated-declarations]
>   ./parallel_hashmap/phmap_base.h:1207:1: warning: 'const_pointer' is deprecated [-Wdeprecated-declarations]
> 

Although my code in tokenizers.bpe is still using parallel-hashmap 1.3.11, I see in the master branch still the use of pointer/const_pointer https://github.com/greg7mdp/parallel-hashmap/blob/master/parallel_hashmap/phmap_base.h#L1236-L1240. 

Is that something which could be fixed? And how?

--- a/parallel_hashmap/phmap_base.h
+++ b/parallel_hashmap/phmap_base.h
@@ -1234,22 +1234,22 @@ using ExtractOrT = typename ExtractOr<Extract, Obj, Default, void>::type;
 
 // Extractors for the features of allocators.
 template <typename T>
-using GetPointer = typename T::pointer;
+using GetPointer = typename std::allocator_traits<T>::pointer;
 
 template <typename T>
-using GetConstPointer = typename T::const_pointer;
+using GetConstPointer = typename std::allocator_traits<T>::const_pointer;
 
 template <typename T>
-using GetVoidPointer = typename T::void_pointer;
+using GetVoidPointer = typename std::allocator_traits<T>::void_pointer;
 
 template <typename T>
-using GetConstVoidPointer = typename T::const_void_pointer;
+using GetConstVoidPointer = typename std::allocator_traits<T>::const_void_pointer;
 
 template <typename T>
-using GetDifferenceType = typename T::difference_type;
+using GetDifferenceType = typename std::allocator_traits<T>::difference_type;
 
 template <typename T>
-using GetSizeType = typename T::size_type;
+using GetSizeType = typename std::allocator_traits<T>::size_type;
 
 template <typename T>
 using GetPropagateOnContainerCopyAssignment =

